lib/ecto/query/api.ex

```diff
@@ -502,6 +502,57 @@ defmodule Ecto.Query.API do
   """
   def splice(list), do: doc! [list]
 
+  @doc """
+  Creates a values list/constant table.
+
+  A values list can be used as a source in a query, both in `Ecto.Query.from/2`
+  and `Ecto.Query.join/5`.
+
+  The first argument is a list of maps representing the values of the constant table.
+  Each entry in the list must have exactly the same fields or an error is raised.
+
+  The second argument is a map of types corresponding to the fields in the first argument.
+  Each field must be given a type or an error is raised. Any type that can be specified in
+  a schema may be used.
+
+  ## Select example
+
+      values = [%{id: 1, text: "abc"}, %{id: 2, text: "xyz"}]
+      types = %{id: :integer}
```

Comment:
Don't we need to give a type for each field? Or do we infer in some cases?

Reply:
sorry that was an oversight. no inference in this PR